sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift

```diff
@@ -0,0 +1,336 @@
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCommunication
+import AzureCore
+import Foundation
+
+public class ChatThreadClient {
+    // MARK: Properties
+
+    public let threadId: String
+    private let client: AzureCommunicationChatClient
+    private let credential: CommunicationUserCredential
+    private let endpoint: String
+    private let options: AzureCommunicationChatClientOptions
+
+    // MARK: Initializers
+
+    /// Create a ChatThreadClient.
+    /// - Parameters:
+    ///   - threadId: The chat thread id.
+    ///   - endpoint: The Communication Services endpoint.
+    ///   - credential: The user credential.
+    ///   - options: Options used to configure the client.
+    public init(
+        threadId: String,
+        endpoint: String,
+        credential: CommunicationUserCredential,
+        withOptions options: AzureCommunicationChatClientOptions
+    ) throws {
+        self.threadId = threadId
+        self.endpoint = endpoint
+        self.credential = credential
+        self.options = options
+
+        guard let baseUrl = URL(string: endpoint) else {
+            throw AzureError.client("Unable to form base URL")
+        }
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        self.client = try AzureCommunicationChatClient(baseUrl: baseUrl, authPolicy: authPolicy, withOptions: options)
+    }
+
+    // MARK: Public Methods
+
+    /// Updates the ChatThread's properties.
+    /// - Parameters:
+    ///   - request: Request for updating a chat thread.
+    ///   - options: Update chat thread options.
+    ///   - completionHandler: A completion handler that receives a status code on success.
+    public func updateThread(
+        request: UpdateChatThreadRequest,
+        withOptions options: UpdateChatThreadOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        client.update(chatThread: request, chatThreadId: threadId, withOptions: options) { result, httpResponse in
```

Comment:
Should mirror the client, omitting the chatThreadId:
```
public func update(thread: UpdateChatThreadRequest, withOptions: ...)
```

Alternatively, you could have the call default to the client's chatThreadId, but let the user override it, but that could get confusing.